contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -3546,3 +3546,262 @@ RETURNS sys.SYSNAME AS $$
 $$
 LANGUAGE plpgsql IMMUTABLE
 STRICT;
+
+
+-- Another defination of date_bucket() to match error messages for unknown date datatype as ANYELEMNT does not support unknown datatype.
+CREATE OR REPLACE FUNCTION sys.date_bucket(IN datepart PG_CATALOG.TEXT, IN number INTEGER, IN date PG_CATALOG.TEXT, IN origin PG_CATALOG.TEXT default NULL) RETURNS PG_CATALOG.TEXT 
+AS 
+$body$
+DECLARE
+BEGIN
+    IF datepart NOT IN ('year', 'quarter', 'month', 'week', 'day', 'hour', 'minute', 'second', 'millisecond') THEN
+            RAISE EXCEPTION '% is not a recognized Date_Bucket option.', datepart;
+    END IF;
+    IF (number IS NULL) THEN
```

Comment:
why are we not using if..elseif..else ?

Reply:
The braces around the if condition are redundant.

Reply:
Resolved